aws_marketplace/using-open-source-model-packages/pytorch-ic-model/notebook.ipynb

```diff
@@ -0,0 +1,461 @@
+    "#### Contents:\n",
+    "1. [Subscribe to the model package](#1.-Subscribe-to-the-model-package)\n",
+    "2. [Create an endpoint and perform real-time inference](#2.-Create-an-endpoint-and-perform-real-time-inference)\n",
+    "   1. [Create an endpoint](#A.-Create-an-endpoint)\n",
+    "   2. [Create input payload](#B.-Create-input-payload)\n",
+    "   3. [Perform real-time inference](#C.-Perform-real-time-inference)\n",
+    "   4. [Visualize output](#D.-Visualize-output)\n",
+    "   5. [Delete the endpoint](#E.-Delete-the-endpoint)\n",
+    "3. [Perform batch inference](#3.-Perform-batch-inference) \n",
+    "4. [Clean-up](#4.-Clean-up)\n",
+    "    1. [Delete the model](#A.-Delete-the-model)\n",
+    "    2. [Unsubscribe to the listing (optional)](#B.-Unsubscribe-to-the-listing-(optional))\n",
+    "    \n",
+    "\n",
+    "#### Usage instructions\n",
+    "You can run this notebook one cell at a time (By using Shift+Enter for running a cell).\n",
+    "\n",
+    "**Note** - This notebook requires you to follow instructions and specify values for parameters, as instructed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#In this notebook, we will be using smart-open package. Let us download the same.\n",
+    "import sys\n",
+    "!{sys.executable} -m pip install smart-open"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Subscribe to the model package"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To subscribe to the model package:\n",
+    "1. Open the model package listing page you opened this notebook for.\n",
+    "1. On the AWS Marketplace listing, click on the **Continue to subscribe** button.\n",
+    "1. On the **Subscribe to this software** page, review and click on **\"Accept Offer\"** if you and your organization agrees with EULA, pricing, and support terms. \n",
+    "1. Once you click on **Continue to configuration button** and then choose a **region**, you will see a **Product Arn** displayed. This is the model package ARN that you need to specify while creating a deployable model using Boto3. Copy the ARN corresponding to your region and specify the same in the following cell."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_package_arn='<Customer to specify Model package ARN corresponding to their AWS region>'         "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from smart_open import smart_open\n",
+    "import json \n",
+    "from sagemaker import ModelPackage\n",
+    "import sagemaker as sage\n",
+    "from sagemaker import get_execution_role\n",
+    "from sagemaker import ModelPackage\n",
+    "import boto3\n",
+    "from IPython.core.display import HTML\n",
```

Comment:
Do you really need this? If we get CI working w/ marketplace notebooks, this will likely cause a failure.

Reply:
Yes, we use sagemaker api create endpoint.

Reply:
I was referring to the HTML package.